Kernel/Devices/Device.cpp

```diff
@@ -15,21 +17,69 @@
 
 namespace Kernel {
 
-Device::Device(MajorNumber major, MinorNumber minor)
-    : m_major(major)
-    , m_minor(minor)
+struct AllDevicesDetails {
+    SpinlockProtected<HashMap<u64, BlockDevice*>, LockRank::None> block_devices {};
+    SpinlockProtected<HashMap<u64, CharacterDevice*>, LockRank::None> char_devices {};
+    SpinlockProtected<CircularQueue<DeviceEvent, 100>, LockRank::None> event_queue {};
+    // NOTE: There's no locking on this pointer because we expect to initialize it once
+    // and never touch it again.
+    OwnPtr<BaseDevices> base_devices;
+};
+
+static Singleton<AllDevicesDetails> s_all_details;
+
+SpinlockProtected<CircularQueue<DeviceEvent, 100>, LockRank::None>& Device::event_queue()
+{
+    return s_all_details->event_queue;
+}
+
+BaseDevices* Device::base_devices()
+{
+    return s_all_details->base_devices.ptr();
+}
+
+UNMAP_AFTER_INIT void Device::initialize_base_devices()
 {
+    auto base_devices = MUST(try_make<BaseDevices>(*NullDevice::must_initialize(), *ConsoleDevice::must_create(), *DeviceControlDevice::must_create()));
```

Comment:
No need to `try_make` this (I think?), just initialize the struct directly.

Reply:
I will check this in a second revision (when I send it).